### PR TITLE
Icon, desktop, appdata and flatpakref fixes

### PR DESCRIPTION
**Pull request by krnowak**
The most important thing here is fixing the desktop file, so firefox shows up on the list of possible default web browser in gnome-control-center and installing an icon, so we do not show an ugly grey icon instead of the firefox logo in various places.

https://phabricator.endlessm.com/T16243

### CODE REVIEW

File: org.mozilla.Firefox.appdata.xml [hunk 1]
```diff
@@ -11,7 +12,7 @@
   <url type="homepage">https://www.mozilla.org/en-US/</url>
   <screenshots>
     <screenshot type="default">
-      <image>http://www.codeblocks.org/images/stories/imagebrowser/scr1.png</image>
+      <image>https://raw.githubusercontent.com/endlessm/firefox-flatpak/master/screenshot.png</image>
```
Comment (joaquimrocha) [inline org.mozilla.Firefox.appdata.xml]: The URL should rather be: https://d3lapyynmdp1i9.cloudfront.net/screenshots/org.mozilla.Firefox/C/org.mozilla.firefox-screenshot1.jpg

File: org.mozilla.Firefox.json [hunk 2]
```diff
@@ -165,6 +165,11 @@
         },
         {
           "type": "file",
+          "url": "https://github.com/endlessm/eos-shell-content/raw/master/icons/bundle/64x64/apps/eos-app-org.mozilla.Firefox.png",
```
Comment (joaquimrocha) [inline org.mozilla.Firefox.json]: The URL should be added to the .in manifest as well. It's not being used right now but we should start using it.

**Review by joaquimrocha — CHANGES_REQUESTED** [review]
Nice fixes @krnowak !
Just a couple of comments and also I think it's a good opportunity to fix this thing of having two json manifests. We should remove the final .json and use the .in. For that we should also replace the _branch_ value by _"@BRANCH@"_ as the Android Studio does.
Can you include these changes?

**Comment by krnowak** [comment]
I have not removed the `.json` file in favor of just keeping `.json.in` only yet. From what I have seen, that `@BRANCH@` thing in android-studio flatpak is for the `runtime-version` field and yet you propose to put it for the `branch` field in firefox manifest. Isn't the `branch` field pointless in the manifest? So, I guess you wanted `"runtime-version": "@BRANCH@"` instead of `"branch": "@BRANCH@"`, which leads me to another question - the `@BRANCH@` substitution would be done by the `jenkins-build-flatpak-manifest` script which produces the file without the `.in` suffix, right? And yet, the `generate-json` script also takes an `.in` file to output the one without it. Do we need a `.in.in` file then there?

Also, regenerating the manifest file every time we fix some typo can be annoying, especially that it involves downloading all those `xpi` files to get their checksums, even when the firefox version did not change.

What I would propose is:
- Remove the `branch` field, the version is already stored in the generate-json script.
- Remove the `org.mozilla.Firefox.json.in` file.
- Modify the `generate-json` script so it changes the manifest file in place. It would simply replace all the `--extra-data` flags in the `finish-args` array with new ones. I'd rename the script then to something like `bump-version`.
- Replace the hard-coded value in the `runtime-version` field with the `@BRANCH@` substitution and rename the manifest file to have the `.in` suffix for the jenkins script to consume.

**Comment by joaquimrocha** [comment]
Yeah, I meant the runtime-version.
The only problem with the approach you're suggesting is that if we want a placeholder in the _.in_ that is just for the //generate_json// (e.g. @MODULE_DEPENDENCIES@), then it could not be an in-place substitution. But we can address that problem when we have it. Let's do what you say please.

**Comment by krnowak** [comment]
Updated.

**Pull request merged**